### DISCUSSION TITLE
Problem with unifi protect not loading after updates to December 2021

### DISCUSSION THREAD

**Discussion by tobes49** [briis/unifiprotect#418] [Q&A] [answered]
Hi

Ive updated my home assistant and unifi protect / integration and now have a problem with it not loading??

Unifi Protect is 1.20.1  
Home Assistant is os7 and 2021.12.4 
Unifi Protect integration is 0.11.0 

IN the logs I have this error:

> Logger: homeassistant.components.camera
> Source: custom_components/unifiprotect/camera.py:83
> Integration: Camera (documentation, issues)
> First occurred: 20:47:37 (1 occurrences)
> Last logged: 20:47:37
> 
> Error while setting up unifiprotect platform for camera
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
>     await asyncio.shield(task)
>   File "/config/custom_components/unifiprotect/camera.py", line 96, in async_setup_entry
>     for camera, channel, is_default in get_camera_channels(data.api):
>   File "/config/custom_components/unifiprotect/camera.py", line 83, in get_camera_channels
>     yield camera, camera.channels[0], True
> IndexError: list index out of range
> 

I have tried reloading the integration
Ive also restarted the UDM-Pro
Deleted and reinstall the unifi protect integration and this finds all the camera and asks for a location but then then fails to load,
Also tried rolling back to 2021.11 but this doesn't solve the problem

Can anyone give me any help towards this please?

**Comment by AngellusMortis** [comment]
What cameras do you have?

Do you not have RTSP enabled on any of your cameras? 

Does it show a high/medium/low option under the Settings -> Advanced section for your camera?

![image](https://user-images.githubusercontent.com/490848/146997176-78478474-772e-437d-9794-785252df0ce5.png)

**Reply by tobes49** [reply]
I have G4 doorbell, 4 xG3, 1xG3 dome and G3 instant
![unfi camera](https://user-images.githubusercontent.com/73558713/146998040-e567843e-bd8d-462d-9cc3-228c989281a0.jpg)
they were all set like this but the doorbell didn't appear to have RTSP - so have turned that on but still not working.

**Reply by AngellusMortis** [reply]
What do you mean "doorbell didn't appear to have RTSP"? Like it does not have that section at all?

**Reply by tobes49** [reply]
> What do you mean "doorbell didn't appear to have RTSP"? Like it does not have that section at all?

 The high, medium and low stream were all turned off for the doorbell,  I've just turned the High slider on

**Reply by AngellusMortis** [reply]
I am trying to understand your problem instead of just working around it. If your cameras do not provide a channel, you cannot stream them, only get still images. So, sorry if this gets too technical.

Can you install the JSONView browser extension (Chrome: https://chrome.google.com/webstore/detail/jsonview/gmegofmjomhknnokphhckolhcffdaihd, Firefox: https://addons.mozilla.org/en-US/firefox/addon/jsonview/) and then go to `https://{protect_ip}/proxy/protect/api/cameras`?

Search for "channels" and just kind of go through each of them (ignore the channels under the talkbackSettings). They should look like this
![image](https://user-images.githubusercontent.com/490848/147001208-98f1d5de-b81c-47c8-aa7f-9b1dd302b6c5.png)


Can you see if you can find the one that does _not_ have any channels?

**Reply by AngellusMortis** [reply]
Also, please make sure the user you are using for Home Assistant has the permission to view all the cameras as well (we really only actively test with the default "Administrator" role)

**Reply by AngellusMortis** [reply]
I also just added `v0.11.2-beta1` which should at least let you load the integration and will now give you a warning log message about the issue. If you can click "Update Information" for the integration and then do "Redownload", check "Show beta versions" and install 0.11.2-beta1

![image](https://user-images.githubusercontent.com/490848/147002034-7ae95a18-3b1d-4dbe-8ad2-273d4e07ca42.png)

That should let the integration load for you, but you will not have a camera entity for any cameras with missing channels. After it loads, if you can go to Configuration -> Settings -> Logs and look for the log message that says something along the lines of the following and post it back here (should also have the name and camera ID).

> Camera does not have any channels:

**Reply by tobes49** [reply]
I will check into this tomorrow,    it was all working correctly (and had been for about a year) and used to stream all my camera feeds to a view on home assistant. This has only stopped with the latest updated.  
Unfortunately I don't know when it went wrong as I updated both protect + integration at the same time as the release notes stated you have to use 1.20 with the latest update.

I will check tomorrow and let you know.  Thank you for the help.

**Reply by AngellusMortis** [reply]
Well, if you can update to 0.11.2-beta1, that should at least get your running again, but you will likely not have streaming for some of your cameras. Then we can troubleshoot the issue tomorrow.

**Reply by tobes49** [reply]
Thank you for the update, here is the log.

> This error originated from a custom integration.
> 
> Logger: custom_components.unifiprotect.camera
> Source: custom_components/unifiprotect/camera.py:35
> Integration: UniFi Protect (documentation, issues)
> First occurred: 22:08:24 (3 occurrences)
> Last logged: 22:08:24
> 
> Camera does not have any channels: None (id: 5f8301050365f703870003eb)
> Camera does not have any channels: None (id: 5f830105038df703870003ee)
> Camera does not have any channels: None (id: 5f8301050383f703870003ed)

**Reply by AngellusMortis** [reply]
Based on the fact there are no names for those cameras, my first guess is you got a bad Protect update. 

These should be the direct URLs to those cameras though if they exist:

https://{protect_ip}/protect/devices/5f8301050365f703870003eb
https://{protect_ip}/protect/devices/5f830105038df703870003ee
https://{protect_ip}/protect/devices/5f8301050383f703870003ed

Just replace `{protect_ip}` with your real protect IP and open it in your browser. If the cameras are wrong or somehow messed up, you might try forgetting them, reset it them and re-add them.

**Reply by tobes49** [reply]
I have a dedicated user for home assistant.  They are a limited admin,  local access and has view only access to protect.
Would I be best to change this to admin as well for protect?

**Reply by tobes49** [reply]
Thank you, I'll try this tomorrow and report back.

> Based on the fact there are no names for those cameras, my first guess is you got a bad Protect update

You mean the unifi protect 1 20.1 update was bad?

**Reply by AngellusMortis** [reply]
Does that user not have access to those 3 cameras? 

If you can, it is probably a good idea. As I mentioned, we do not test against any of the other roles exception for the default Administrator one (not View Only). If more we keep running into people having issues with it, we can set up more automated testing to verify nothing breaks with specific roles.

**Reply by AngellusMortis** [reply]
> You mean the unifi protect 1 20.1 update was bad?

Yeah, if UniFi Protect is reporting 3 cameras with no name, it sounds like a UniFi Protect issue.

**Reply by AngellusMortis** [reply]
Someone else reported a similar issue. So, lets move the discussion over there: https://github.com/briis/unifiprotect/issues/420

**Reply by AngellusMortis** [reply]
We have confirmed the issue. Full details here: https://github.com/briis/unifiprotect/issues/420

Fix will be part of 0.11.2 (PR to fix: https://github.com/briis/unifiprotect/pull/421)

**Comment by AngellusMortis** [comment, marked as answer]
We have confirmed the issue. Full details here: https://github.com/briis/unifiprotect/issues/420

Fix will be part of 0.11.2 (PR to fix: https://github.com/briis/unifiprotect/pull/421)
